src/main/serialport.ts

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import type { AutoDetectTypes, SetOptions } from '@serialport/bindings-cpp';
+import { Renderer, WebContents } from 'electron';
+import { SERIALPORT_CHANNEL } from 'pc-nrfconnect-shared/main';
+import { SerialPort, SerialPortOpenOptions } from 'serialport';
+
+export type Renderer = Pick<WebContents, 'on' | 'send' | 'id'>;
+
+/**
+ *  serialPorts: {
+
+        '/dev/Robot': {
+            renderers: Renderer[],
+            serialPort: SerialPort,
+        },
+
+        '/dev/Rowboat': {
+            renderers: Renderer[],
+            serialPort: SerialPort,
+        }
+    }
+*/
+type OpenPort = {
+    serialPort: SerialPort;
+    renderers: Renderer[];
+    settingsLocked: boolean;
+    opening: boolean;
+};
+
+const serialPorts = new Map<string, OpenPort>();
+
+export const openOrAdd = async (
+    sender: Renderer,
+    options: SerialPortOpenOptions<AutoDetectTypes>,
+    overwrite: boolean
+) => {
+    const { path } = options;
+    const existingPort = serialPorts.get(path);
+
+    if (existingPort) {
+        // Port is already opened
+
+        if (existingPort.opening) {
+            throw new Error('PORT_IS_ALREADY_BEING_OPENED');
+        }
+
+        const renderers = existingPort.renderers;
+        const alreadyInList = renderers.find(
+            existing => sender.id === existing.id
+        );
+
+        if (!alreadyInList) {
+            let isDifferentSettings;
```

Comment:
I've changed the instantiation to use `some` instead of a filter, then it should be explicit that the variable should hold a boolean value, meaning either the sender is already in the list, or it is not.